statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,245 @@
+        self.communality = None
+        self.eigenvals = None
+        super(Factor, self).__init__(endog, exog)
+
+    def fit(self, n_max_iter=50, tolerance=1e-6, rotation=None,
+            SMC=True):
+        """
+        Fit the factor model
+
+        Parameters
+        ----------
+        n_max_iter : int
+            Maximum number of iterations for communality estimation
+        tolerance : float
+            If `norm(communality - last_communality)  < tolerance`,
+            estimation stops
+        rotation : string
+            rotation to be applied
+        SMC : True or False
+            Whether or not to apply squared multiple correlations
+
+        -------
+
+        """
+        if rotation not in [None, 'varimax', 'quartimax', 'biquartimax',
+                            'equamax', 'oblimin', 'parsimax', 'parsimony',
+                            'biquartimin', 'promax']:
+            raise ValueError('Unknown rotation method %s' % (rotation))
+        R = pd.DataFrame(self.endog).corr().values
+        self.n_comp = matrix_rank(R)
+        if self.n_factor > self.n_comp:
+            raise ValueError('n_factor must be smaller or equal to the rank'
+                             ' of endog! %d > %d' %
+                             (self.n_factor, self.n_comp))
+        if n_max_iter <= 0:
+            raise ValueError('n_max_iter must be larger than 0! %d < 0' %
+                             (n_max_iter))
+        if tolerance <= 0 or tolerance > 0.01:
+            raise ValueError('tolerance must be larger than 0 and smaller than'
+                             ' 0.01! Got %f instead' % (tolerance))
+        #  Initial communality estimation
+        if SMC:
+            c = 1 - 1 / np.diag(inv(R))
+            self.SMC = np.array(c)
+        else:
+            c = np.ones([1, len(R)])
+
+        # Iterative communality estimation
+        eigenvals = None
+        for i in range(n_max_iter):
+            # Get eigenvalues/eigenvectors of R with diag replaced by
+            # communality
+            for j in range(len(R)):
+                R[j, j] = c[j]
+            L, V = eig(R)
+            c_last = np.array(c)
+            ind = np.argsort(L)
+            ind = ind[::-1]
+            L = L[ind]
+            n_pos = (L > 0).sum()
+            V = V[:, ind]
+            eigenvals = np.array(L)
+
+            # Select eigenvectors with positive eigenvalues
+            n = np.min([n_pos, self.n_factor])
+            sL = np.diag(np.sqrt(L[:n]))
+            V = V[:, :n]
+
+            # Calculate new loadings and communality
+            A = V.dot(sL)
+            c = np.power(A, 2).sum(axis=1)
+            if norm(c_last - c) < tolerance:
+                break
+        self.eigenvals = eigenvals
+        self.communality = c
+        # Perform rotation of the loadings
+        self.loadings_no_rot = np.array(A)
+        if rotation in ['varimax', 'quartimax', 'biquartimax', 'equamax',
```

Comment:
possible structural change:  given that we can use different rotation with same raw FA
use pattern as for cov_type
- make rotation a fit keyword
- set the rotation in the results class `__init__`
- try to avoid any rotation specific attributes in `Factor` and attach those to results instance

to get a new results instance we could make the optimization loop conditional on not having `loadings_not_rot` as Factor attribute. 

maybe rename `loadings_not_rot` -> `loadings_raw` (or just `loadings` if it's unambiguous in the `Factor` class. (while FactorResults has the rotated loading).